editor/src/components/editor/component-button.tsx

```diff
@@ -0,0 +1,202 @@
+/** @jsx jsx */
+import { jsx } from '@emotion/react'
+import * as React from 'react'
+import { BreadcrumbTrail } from '../canvas/controls/breadcrumb-trail'
+import {
+  FlexRow,
+  Icn,
+  OnClickOutsideHOC,
+  SmallerIcons,
+  UtopiaTheme,
+} from '../../uuiui'
+import { betterReactMemo } from '../../uuiui-deps'
+import { NameRowCrumbs } from '../canvas/controls/namerowcrumbs'
+
+export const ComponentOrInstanceIndicator = betterReactMemo(
+  'ComponentOrInstanceIndicator',
+  () => {
+    const [isOpen, setIsOpen] = React.useState(false)
+    const toggleOpen = React.useCallback(() => {
+      setIsOpen(!isOpen)
+    }, [isOpen])
+
+    const closeAndEatEvent = React.useCallback(
+      (e: MouseEvent) => {
+        setIsOpen(false)
+        e.stopPropagation()
+        e.preventDefault()
+      },
+      [setIsOpen],
+    )
+
+    enum editContext {
+      ELEMENT = 0,
+      COMPONENT_INSTANCE,
+      COMPONENT,
+      COMPONENT_CHILD,
+    }
+
+    const [currentEditContext, setCurrentEditContext] = React.useState(editContext.COMPONENT)
+
+    const TODOREPLACEMEcycleEditContext = () => {
```

Comment:
I see you come from the @balazsbajorics school of naming conventions :grinning:

Reply:
(this function is the one that will need to be replaced entirely once we're able to "focus" components)

Reply:
this is the way.